File: app/src/main/java/com/lucasurbas/counter/service/usecase/GetCounterInteractor.java
package com.lucasurbas.counter.service.usecase;

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers;
import com.lucasurbas.counter.data.model.Counter;
import com.lucasurbas.counter.data.repository.CounterRepository;

import javax.inject.Inject;

import io.reactivex.Single;

public class GetCounterInteractor {

    private final CounterRepository counterRepository;
    private final InteractorSchedulers interactorSchedulers;

    @Inject
    public GetCounterInteractor(CounterRepository counterRepository,
                                               InteractorSchedulers interactorSchedulers){
        this.counterRepository = counterRepository;
        this.interactorSchedulers = interactorSchedulers;
    }

    public Single<Counter> getCounter(int counterId){
        return counterRepository.getCounterWithId(counterId)
                .subscribeOn(interactorSchedulers.getBackgroundScheduler())
                .observeOn(interactorSchedulers.getMainThreadScheduler());
    }
}

File: app/src/main/java/com/lucasurbas/counter/service/usecase/GetCounterInteractor.kt
package com.lucasurbas.counter.service.usecase

import com.lucasurbas.counter.app.di.rx.InteractorSchedulers
import com.lucasurbas.counter.data.model.Counter
import com.lucasurbas.counter.data.repository.CounterRepository
import io.reactivex.Single
import javax.inject.Inject

class GetCounterInteractor @Inject constructor(
        private val counterRepository: CounterRepository,
        private val interactorSchedulers: InteractorSchedulers) {

    fun getCounter(counterId: Int): Single<Counter> {
        return counterRepository.getCounterWithId(counterId)
                .subscribeOn(interactorSchedulers.backgroundScheduler)
                .observeOn(interactorSchedulers.mainThreadScheduler)
    }
}
